# Principal Component Analysis (PCA)

In [3]:
import pandas as pd
import prince

%matplotlib inline

For this notebook we will be using bike sharing data from the city of Toulouse.

In [4]:
df = pd.read_csv('data/bike_stations.csv', index_col=0)
df.head(5)

0-0       1-0       2-0       3-0       4-0  \
00189 - MURET BOUTINON      0.890442  0.783402  0.754204  0.694712  0.644286   
00157 - U.R.S.S  PEYROUSET  0.794423  0.694662  0.666052  0.611976  0.727828   
00069 - FEUGA               0.237852  0.246162  0.339617  0.288618  0.334571   
00181 - PLANA GARNIER       0.677826  0.659270  0.711563  0.626856  0.620313   
00014 - PERIGORD            0.377450  0.221058  0.165181  0.175972  0.234515   

                                 5-0       6-0       0-1       1-1       2-1  \
00189 - MURET BOUTINON      0.520034  0.593616  0.899294  0.828925  0.796468   
00157 - U.R.S.S  PEYROUSET  0.685627  0.744462  0.796654  0.734205  0.721514   
00069 - FEUGA               0.341950  0.300178  0.251663  0.240979  0.348627   
00181 - PLANA GARNIER       0.676800  0.536205  0.691771  0.671441  0.712834   
00014 - PERIGORD            0.299102  0.425212  0.324483  0.198676  0.123316   

                               ...          0-23      1-23      2-23  \
00189 - MURET BOUTINON         ...      0.764102  0.675706  0.656017   
00157 - U.R.S.S  PEYROUSET     ...      0.682245  0.617044  0.673086   
00069 - FEUGA                  ...      0.274072  0.321278  0.310949   
00181 - PLANA GARNIER          ...      0.638449  0.674415  0.595292   
00014 - PERIGORD               ...      0.241303  0.166612  0.218694   

                                3-23      4-23      5-23      6-23   latitude  \
00189 - MURET BOUTINON      0.562182  0.501754  0.550409  0.861733  43.581352   
00157 - U.R.S.S  PEYROUSET  0.697583  0.673767  0.717911  0.786854  43.582832   
00069 - FEUGA               0.376079  0.406024  0.335113  0.251404  43.592530   
00181 - PLANA GARNIER       0.612726  0.681631  0.556120  0.673393  43.610549   
00014 - PERIGORD            0.262935  0.294839  0.476275  0.371588  43.607352   

                            longitude    altitude  
00189 - MURET BOUTINON       1.425326  145.218811  
00157 - U.R.S.S  PEYROUSET   1.448634  149.000000  
00069 - FEUGA                1.442900  145.611435  
00181 - PLANA GARNIER        1.479500  141.108444  
00014 - PERIGORD             1.444270  150.409821  

[5 rows x 171 columns]

Each cell contains the average $\frac{bikes}{bikes + spaces}$ ratio. Each column name is a weekday-hour pair (with weekdays starting at 0). For example, the *00189 - Muret Boutinon* station is on average 89% full on Mondays at midnight (first row, first column). 

The last three columns contain information about the location of each station (latitude, longitude and altitude).

In [6]:
pca = prince.PCA(df)

OSError: 'prince/plot/mpl/prince.mplstyle' not found in the style library and input is not a valid URL or path. See `style.available` for list of available styles.